### PR TITLE
Add support for finding executable name on bsd

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

File: port/unix/omrsysinfo.c [hunk 1]
```diff
@@ -1705,6 +1707,22 @@ find_executable_name(struct OMRPortLibrary *portLibrary, char **result)
...
 	return rc;
+#elif defined(OMR_BSD)
+	int proc[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+	size_t size = 0;
+	if (0 == sysctl(proc, sizeof(proc) / sizeof(proc[0]), NULL, &size, NULL, 0)) {
+		*result = portLibrary->mem_allocate_memory(portLibrary, size, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+	} else {
+
+		return -1;
+	}
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]: If you're going to be using multiple returns, I think this is better structured as:
```
if (0 != sysctl(...)) {
  return -1;
}

*result = portlibrary->mem_allocate(...);
```

Just like you did in the next call below.

And, don't forget to check if the new allocation worked.

File: port/unix/omrsysinfo.c [hunk 2]
```diff
@@ -1705,7 +1707,25 @@ find_executable_name(struct OMRPortLibrary *portLibrary, char **result)
...
-#else /* defined(OSX) */
+#elif defined(OMR_BSD)
+	int proc[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+	size_t size = 0;
+	if (-1 == sysctl(proc, sizeof(proc) / sizeof(proc[0]), NULL, &size, NULL, 0)) {
+		return -1;
+	}
+
+	*result = portLibrary->mem_allocate_memory(portLibrary, size, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+	if(NULL == *result){
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]:
```suggestion
	if(NULL == *result) {
```

File: port/unix/omrsysinfo.c [hunk 3]
```diff
@@ -43,10 +43,12 @@
...
 #include <sys/param.h>
 #include <sys/mount.h>
+#endif /* defined(LINUX) && !defined(OMRZTPF) */
+
+#if defined(OSX) || defined(OMR_BSD)
 #include <sys/types.h>
 #include <sys/sysctl.h>
-
-#endif /* defined(LINUX) && !defined(OMRZTPF) */
+#endif() /* defined(OSX) || defined(OMR_BSD) */
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]:
```suggestion
#endif /* defined(OSX) || defined(OMR_BSD) */
```

File: port/unix/omrsysinfo.c [hunk 4]
```diff
@@ -1705,7 +1707,25 @@ find_executable_name(struct OMRPortLibrary *portLibrary, char **result)
...
+	}
+
+	*result = portLibrary->mem_allocate_memory(portLibrary, size, OMR_GET_CALLSITE(), OMRMEM_CATEGORY_PORT_LIBRARY);
+	if(NULL == *result){
+		return -1;
+	}
+
+	if (-1 == sysctl(proc, sizeof(proc) / sizeof(proc[0]), *result, &size, NULL, 0)) {
+		portLibrary->mem_free_memory(portLibrary, *result);
+		*result = 0;
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]:
```suggestion
		*result = NULL;
```

File: port/unix/omrsysinfo.c [hunk 5]
```diff
@@ -1705,7 +1707,25 @@ find_executable_name(struct OMRPortLibrary *portLibrary, char **result)
 		rc = 0;
 	}
 	return rc;
-#else /* defined(OSX) */
+#elif defined(OMR_BSD)
+	int proc[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+	size_t size = 0;
+	if (-1 == sysctl(proc, sizeof(proc) / sizeof(proc[0]), NULL, &size, NULL, 0)) {
+		return -1;
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]: should probably null out result here, before returning.

File: port/unix/omrsysinfo.c (hunk 3 above)
Comment (dnakamura) [reply port/unix/omrsysinfo.c]: ugh. I think I've spent too much time in cmake  😄

**Comment by youngar** [comment]
@genie-omr build all

File: port/unix/omrsysinfo.c [hunk 6]
```diff
@@ -1705,7 +1707,26 @@ find_executable_name(struct OMRPortLibrary *portLibrary, char **result)
...
+	}
+
+	if (-1 == sysctl(proc, sizeof(proc) / sizeof(proc[0]), *result, &size, NULL, 0)) {
+		portLibrary->mem_free_memory(portLibrary, *result);
+		*result = NULL;
+		return -1;
+	}
+	return 0;
+#else /* defined(OMR_BSD) */
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]:
```suggestion
#else /* defined(OMR_OS_BSD) */
```

File: port/unix/omrsysinfo.c [hunk 7]
```diff
@@ -43,10 +43,12 @@
 #include <mach-o/dyld.h>
 #include <sys/param.h>
 #include <sys/mount.h>
+#endif /* defined(LINUX) && !defined(OMRZTPF) */
+
+#if defined(OSX) || defined(OMR_BSD)
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]:
```suggestion
#if defined(OSX) || defined(OMR_OS_BSD)
```

File: port/unix/omrsysinfo.c [hunk 8]
```diff
@@ -43,10 +43,12 @@
...
 #include <sys/types.h>
 #include <sys/sysctl.h>
-
-#endif /* defined(LINUX) && !defined(OMRZTPF) */
+#endif /* defined(OSX) || defined(OMR_BSD) */
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c]:
```suggestion
#endif /* defined(OSX) || defined(OMR_OS_BSD) */
```

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build win

**Pull request merged**